datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/AbstractBloomDataMapWriter.java

```diff
@@ -281,4 +287,13 @@ protected void releaseResouce() {
     }
   }
 
+  /**
+   * BloomDataMapBuilder(called when datamap rebuild) is set to true;
+   * BloomDataMapWriter(called when data load) is set to false;
+   *
+   * The reason for this is dict index column is already decoded to surrogate key when rebuild
+   * but it is still byte array if build datamap when loading
+   */
+  abstract boolean isRebuildProcess();
```

Comment:
It is better to abstract the difference of behavior instead of adding this.

Reply:
fix by abstract  method `convertDictionaryValue`